### PR TITLE
Make supervisor_export_image test image content.

### PR DESCRIPTION
**Pull request by brettle**
Use exported image as a texture and do a basic check for correctness.

I'm also experiencing some strange behavior with image export on my local machine and this is a way to figure out whether it also happens during CI.

### CODE REVIEW

**Review by stefaniapedrazzi — COMMENTED** [review]
It is good to also test the content of the exported image.
Camera image colors might slightly differ due to the differences in the graphics card.
The test is still failing because of the move handles that are displayed for the selected object.
The selection should be disabled from the supervisor by selecting for example the Viewpoint or the WorldInfo node or it could be set in the WBPROJ file (that in this particular case will have to be committed to the repo).
In the WBPROJ file you simply have to add the line:
```
userInteractions: selectionDisabled
```

Meanwhile, I have a couple of comments about the coding style.

**Comment by brettle** [comment]
Thanks for the feedback.

Why would color differ by graphics card in this case? For the image I'm exporting, I'm using a material that emits the pure color and for the image grabbed by the camera, I'm using such a bright light (with no attenuation) that the image should be saturated. If it's not, I can increase the light intensity until it is.

**Comment by brettle** [comment]
Any idea why this is still failing on ubuntu-20.04? The exported image which is supposed be green is actually the color of the sky. I added additional planes to completely enclose the viewpoint (in colors that aren't the blue of the sky) and I changed the workflow to make the `tests/` directory available as an artifact if the test suite fails. The `GREEN.png` file from that artifact should be green but is this instead:

GREEN

File: tests/with_rendering/controllers/supervisor_export_image/supervisor_export_image.c [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+#include "../../../lib/ts_assertion.h"
+#include "../../../lib/ts_utils.h"
+
+#include <math.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#define FILE_NAME_PNG "image.png"
+#define FILE_NAME_JPG "image.jpg"
+
```
Comment (omichel) [inline tests/with_rendering/controllers/supervisor_export_image/supervisor_export_image.c:12]: This doesn't seem to be used.
```suggestion
```

File: tests/with_rendering/controllers/supervisor_export_image/supervisor_export_image.c [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+bool file_exists(const char *filename) {
```
Comment (omichel) [inline tests/with_rendering/controllers/supervisor_export_image/supervisor_export_image.c:13]:
```suggestion
static bool file_exists(const char *filename) {
```

File: tests/with_rendering/controllers/supervisor_export_image/supervisor_export_image.c [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  remove(file_name_png);
+  remove(file_name_jpg);
+
+  ts_assert_boolean_not_equal(file_exists(file_name_png), "Impossible to remove the PNG file");
+  ts_assert_boolean_not_equal(file_exists(file_name_jpg), "Impossible to remove the JPG file");
+
+  // Turn to face the requested direction
+  WbFieldRef rotation_field = wb_supervisor_node_get_field(wb_supervisor_node_get_from_def("Test"), "rotation");
+  double rotation[4] = {0.0, 0.0, 1.0, 0.0};
+  rotation[3] = angle_radians;
```
Comment (omichel) [inline tests/with_rendering/controllers/supervisor_export_image/supervisor_export_image.c:42]:
```suggestion
  double rotation[4] = {0.0, 0.0, 1.0, angle_radians};
```

File: tests/with_rendering/worlds/supervisor_export_image.wbt [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+Background {
+  skyColor [
+    0.4 0.7 1
+  ]
+}
+PointLight {
+  ambientIntensity 1
+  intensity 10
+}
+DEF Test Robot {
```
Comment (omichel) [inline tests/with_rendering/worlds/supervisor_export_image.wbt:24]: We generally use all caps for DEF names.
```suggestion
DEF TEST Robot {
```

File: tests/with_rendering/controllers/supervisor_export_image/supervisor_export_image.c [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  sprintf(file_name_jpg, "%s.jpg", color_def);
+
+  remove(file_name_png);
+  remove(file_name_jpg);
+
+  ts_assert_boolean_not_equal(file_exists(file_name_png), "Impossible to remove the PNG file");
+  ts_assert_boolean_not_equal(file_exists(file_name_jpg), "Impossible to remove the JPG file");
+
+  // Turn to face the requested direction
+  WbFieldRef rotation_field = wb_supervisor_node_get_field(wb_supervisor_node_get_from_def("Test"), "rotation");
```
Comment (omichel) [inline tests/with_rendering/controllers/supervisor_export_image/supervisor_export_image.c:40]:
```suggestion
  WbFieldRef rotation_field = wb_supervisor_node_get_field(wb_supervisor_node_get_from_def("TEST"), "rotation");
```

**Review by omichel — CHANGES_REQUESTED** [review]

**Review by omichel — APPROVED** [review]
Thank you.

**Pull request merged**